modules/calib3d/src/solvepnp.cpp

```diff
@@ -311,19 +311,23 @@ bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
 
     opoints_inliers.resize(npoints1);
     ipoints_inliers.resize(npoints1);
-    result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
-                      distCoeffs, rvec, tvec, useExtrinsicGuess,
-                      (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    try
+    {
+        result = solvePnP(opoints_inliers, ipoints_inliers, cameraMatrix,
+                          distCoeffs, rvec, tvec, useExtrinsicGuess,
+                          (flags == SOLVEPNP_P3P || flags == SOLVEPNP_AP3P) ? SOLVEPNP_EPNP : flags) ? 1 : -1;
+    }
+    catch (const cv::Exception& e)
+    {
+        result = 0;
+        CV_UNUSED(e);
+        CV_LOG_DEBUG(NULL, "solvePnPRansac(): solvePnP stage failed, use result from MSS(Minimal Sample Sets) stage instead:\n" << e.what());
```

Comment:
Maybe something like this? This is to explain to the user that everything went fine (consensus set was found that agree with the hypothesis), but issue occurs when computing the final pose using all the points in the consensus set.

```suggestion
        CV_LOG_DEBUG(NULL, "solvePnPRansac(): solvePnP stage to compute the final pose using points in the consensus set failed, use result from MSS (Minimal Sample Sets) stage instead:\n" << e.what());
```